frontend/src/app/main/ui/workspace/textpalette.cljs

```diff
@@ -0,0 +1,153 @@
+
+        on-right-arrow-click
+        (mf/use-callback
+         (fn []
+           (when-let [node (mf/ref-val container)]
+             (.scrollBy node #js {:left 200 :behavior "smooth"}))))
+
+        on-wheel
+        (mf/use-callback
+         (fn [event]
+           (let [delta (+ (.. event -nativeEvent -deltaY) (.. event -nativeEvent -deltaX))]
+             (if (pos? delta)
+               (on-right-arrow-click)
+               (on-left-arrow-click)))))
+
+        {:keys [on-pointer-down on-lost-pointer-capture on-mouse-move parent-ref size]}
+        (use-resize-hook :palette 72 54 80 :y true :bottom)]
+    
+    [:div.color-palette {:ref parent-ref
+                         :class (dom/classnames :no-text (< size 72))
+                         :style #js {"--height" (str size "px")}}
+     [:div.resize-area {:on-pointer-down on-pointer-down
+                        :on-lost-pointer-capture on-lost-pointer-capture
+                        :on-mouse-move on-mouse-move}]
+     [:& dropdown {:show (:show-menu @state)
+                   :on-close #(swap! state assoc :show-menu false)}
+
+      [:ul.workspace-context-menu.palette-menu
+       (for [[idx cur-library] (map-indexed vector (vals shared-libs))]
+         (let [typographies (-> cur-library (get-in [:data :typographies]) vals)]
+           [:li.palette-library
+            {:key (str "library-" idx)
+             :on-click #(reset! selected (:id cur-library))}
+
+            (when (= @selected (:id cur-library)) i/tick)
+
+            [:div.library-name (str (:name cur-library) " " (str/format "(%s)" (count typographies)))]]))
+
+       [:li.palette-library
+        {:on-click #(reset! selected :file)}
+        (when (= selected :file) i/tick)
+        [:div.library-name (str (tr "workspace.libraries.colors.file-library")
+                                (str/format " (%s)" (count file-typographies)))]]]]
+
+     [:div.color-palette-actions
+      {:on-click #(swap! state assoc :show-menu true)}
+      [:div.color-palette-actions-button i/actions]]
+
+     [:span.left-arrow {:on-click on-left-arrow-click} i/arrow-slide]
+
+     [:div.color-palette-content {:ref container :on-wheel on-wheel}
+      [:div.color-palette-inside
+       (for [[idx item] (map-indexed vector current-typographies)]
+         [:& typography-item
+          {:key idx
+           :file-id file-id
+           :local local
+           :selected-ids selected-ids
+           :typography item}])]]
+
+     [:span.right-arrow {:on-click on-right-arrow-click} i/arrow-slide]]))
+
+(def local-data
+  (l/derived #(select-keys % [:editors]) refs/workspace-local =))
+
+(mf/defc textpalette
+  {::mf/wrap [mf/memo]}
+  []
+  (let [local             (mf/deref local-data)
+        selected-ids      (mf/deref refs/selected-shapes)
+        file-typographies (mf/deref refs/workspace-file-typography)
+        shared-libs       (mf/deref refs/workspace-libraries)
+        current-file-id   (mf/deref refs/current-file-id)]
```

Comment:
you don't need this deref. You already can use `(mf/use-ctx app.main.ui.context/current-file-id)`

Because workspace already provides this values using react context:

```clojure
    [:& (mf/provider ctx/current-file-id) {:value (:id file)}
     [:& (mf/provider ctx/current-team-id) {:value (:team-id project)}
      [:& (mf/provider ctx/current-project-id) {:value (:id project)}
       [:& (mf/provider ctx/current-page-id) {:value page-id}
       ...
```

Reply:
OK